Libraries/Opc.Ua.Client/Session.cs

```diff
@@ -44,7 +45,7 @@ namespace Opc.Ua.Client
     /// <summary>
     /// Manages a session with a server.
     /// </summary>
-    public partial class Session : SessionClient, IDisposable
+    public partial class Session : SessionClientBatched, IDisposable
```

Comment:
NodeCache object creation needs to be moved from Initialize() method into 
```
private void Initialize(
            ITransportChannel channel,
            ApplicationConfiguration configuration,
            ConfiguredEndpoint endpoint,
            X509Certificate2 clientCertificate)
```
after the block:
```
if (messageContext != null)
{
    m_namespaceUris = messageContext.NamespaceUris;
    m_serverUris = messageContext.ServerUris;
    m_factory = messageContext.Factory;
}
else
{
    m_namespaceUris = new NamespaceTable();
    m_serverUris = new StringTable();
    m_factory = new EncodeableFactory(EncodeableFactory.GlobalFactory);
}
```
because otherwise the NamespaceUris and ServerUris will not be shared with the NodeCache and some nodes might not be accessible for searches in node cache since initially the NamespaceUris contains only namespace 0 and cannot resolve other possible namespace uris that are loaded into the session.

Reply:
@OctaviaStan good catch! fixed!